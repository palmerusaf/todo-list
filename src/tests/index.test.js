import userEvent from "@testing-library/user-event";
import "../index";

it("todo list and footer renders correctly", () => {
  expect(document.body).toMatchSnapshot();
});

it("New project form is rendered correctly", async () => {
  const addProjectButton = document.querySelector(
    ".project-field__add > button"
  );
  await userEvent.click(addProjectButton);
  expect(document.body).toMatchSnapshot();
});

it("New project is created correctly", async () => {
  const form = document.querySelector(".project-field__form");
  const input = form.querySelector("input");
  await userEvent.type(input, "Project one");

  const submit = form.querySelector("button");
  await userEvent.click(submit);
  expect(document.body).toMatchSnapshot();
});

it("Project active class set correctly", () => {
  const projects = document.querySelectorAll(".project-field__item");

  expect(projects[0].classList.toString()).toEqual(
    "project-field__item project-field__item--active"
  );
  expect(projects[1].classList.toString()).toEqual("project-field__item");
});

it("Project is selected correctly", async () => {
  let projects = document.querySelectorAll(".project-field__item");

  await userEvent.click(projects[1]);

  projects = document.querySelectorAll(".project-field__item");

  expect(projects[1].classList.toString()).toEqual(
    "project-field__item project-field__item--active"
  );
  expect(projects[0].classList.toString()).toEqual("project-field__item");
  expect(document.body).toMatchSnapshot();
});

it("New project is updated correctly", async () => {
  const projects = document.querySelectorAll(".project-field__item");
  const editButton = projects[1].querySelector(".button__edit");
  await userEvent.click(editButton);

  const form = document.querySelector(".project-field__form");
  const input = form.querySelector("input");
  await userEvent.clear(input);
  await userEvent.type(input, "Changed Project");

  const submit = form.querySelector("button");
  await userEvent.click(submit);

  expect(document.body).toMatchSnapshot();
});

it("New project is deleted correctly", async () => {
  const projects = document.querySelectorAll(".project-field__item");
  const deleteButton = projects[1].querySelector(".button__delete");
  await userEvent.click(deleteButton);
  await userEvent.click(projects[0]);

  expect(document.body).toMatchSnapshot();
});

it("New task is created correctly", async () => {
  const addButton = document.querySelector(".task-field__add-button");
  await userEvent.click(addButton);

  const title = document.querySelector("input[name='title']");
  const dueDate = document.querySelector("input[name='due-date']");
  const description = document.querySelector("input[name='description']");
  const prioritySel = document.querySelector(".task-field__selector");
  const submit = document.querySelector(".task-form__submit");

  await userEvent.type(title, "Task One Title");
  await userEvent.type(description, "Task One Description");
  await userEvent.type(dueDate, "2020-11-09");
  await userEvent.selectOptions(prioritySel, "Low");

  await userEvent.click(submit);

  expect(document.body).toMatchSnapshot();
});

it("New task is updated correctly", async () => {
  const editButton = document.querySelector(
    ".task-field__button-field > .button__edit"
  );
  await userEvent.click(editButton);
  expect(document.body).toMatchSnapshot();
});

it.todo("New task is deleted correctly");
