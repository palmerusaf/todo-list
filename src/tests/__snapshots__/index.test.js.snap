// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New project form is rendered correctly 1`] = `
<body>
  <div
    class="to-do-list"
  >
    <div
      class="project-field"
    >
      <button
        class="material-icons-outlined button button__arrow_back_ios project-field__expand-button"
      >
        arrow_back_ios
      </button>
      <span
        class="project-field__item project-field__item--active"
        data-project-index="0"
      >
        <span
          class="project-field__title"
        >
          My Project
        </span>
        <span
          class="project-field__buttons"
        >
          <button
            class="material-icons-outlined button button__edit"
          >
            edit
          </button>
          <button
            class="material-icons-outlined button button__delete"
          >
            delete
          </button>
        </span>
      </span>
      <span
        class="project-field__item"
        data-project-index="undefined"
      >
        <form
          class="project-field__form"
        >
          <input
            class="project-field__text-box"
            placeholder="Enter project name"
            required=""
          />
          <button
            class="material-icons-outlined button button__check"
          >
            check
          </button>
        </form>
      </span>
      <span
        class="project-field__item project-field__add"
      >
        <button
          class="material-icons-outlined button button__add"
        >
          add
        </button>
      </span>
    </div>
    <div
      class="task-field"
      data-project-index="0"
    >
      <h1
        class="task-field__project-title"
      >
        Task List for My Project
      </h1>
      <div
        class="task-field__task-list"
      />
      <button
        class="material-icons-outlined button button__add task-field__add-button"
      >
        add
      </button>
    </div>
  </div>
  <footer
    class="footer"
  >
    <a
      class="footer__link"
      href="https://github.com/palmerusaf/todo-list"
    >
      GitHub Repo
    </a>
    <img
      alt="git hub logo"
      class="footer__logo"
      src="https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg"
    />
  </footer>
</body>
`;

exports[`New project is created correctly 1`] = `
<body>
  <div
    class="to-do-list"
  >
    <div
      class="project-field"
    >
      <button
        class="material-icons-outlined button button__arrow_back_ios project-field__expand-button"
      >
        arrow_back_ios
      </button>
      <span
        class="project-field__item project-field__item--active"
        data-project-index="0"
      >
        <span
          class="project-field__title"
        >
          My Project
        </span>
        <span
          class="project-field__buttons"
        >
          <button
            class="material-icons-outlined button button__edit"
          >
            edit
          </button>
          <button
            class="material-icons-outlined button button__delete"
          >
            delete
          </button>
        </span>
      </span>
      <span
        class="project-field__item"
        data-project-index="1"
      >
        <span
          class="project-field__title"
        >
          Project one
        </span>
        <span
          class="project-field__buttons"
        >
          <button
            class="material-icons-outlined button button__edit"
          >
            edit
          </button>
          <button
            class="material-icons-outlined button button__delete"
          >
            delete
          </button>
        </span>
      </span>
      <span
        class="project-field__item project-field__add"
      >
        <button
          class="material-icons-outlined button button__add"
        >
          add
        </button>
      </span>
    </div>
    <div
      class="task-field"
      data-project-index="0"
    >
      <h1
        class="task-field__project-title"
      >
        Task List for My Project
      </h1>
      <div
        class="task-field__task-list"
      />
      <button
        class="material-icons-outlined button button__add task-field__add-button"
      >
        add
      </button>
    </div>
  </div>
  <footer
    class="footer"
  >
    <a
      class="footer__link"
      href="https://github.com/palmerusaf/todo-list"
    >
      GitHub Repo
    </a>
    <img
      alt="git hub logo"
      class="footer__logo"
      src="https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg"
    />
  </footer>
</body>
`;

exports[`New project is deleted correctly 1`] = `
<body>
  <div
    class="to-do-list"
  >
    <div
      class="project-field"
    >
      <button
        class="material-icons-outlined button button__arrow_back_ios project-field__expand-button"
      >
        arrow_back_ios
      </button>
      <span
        class="project-field__item project-field__item--active"
        data-project-index="0"
      >
        <span
          class="project-field__title"
        >
          My Project
        </span>
        <span
          class="project-field__buttons"
        >
          <button
            class="material-icons-outlined button button__edit"
          >
            edit
          </button>
          <button
            class="material-icons-outlined button button__delete"
          >
            delete
          </button>
        </span>
      </span>
      <span
        class="project-field__item project-field__add"
      >
        <button
          class="material-icons-outlined button button__add"
        >
          add
        </button>
      </span>
    </div>
    <div
      class="task-field"
      data-project-index="0"
    >
      <h1
        class="task-field__project-title"
      >
        Task List for My Project
      </h1>
      <div
        class="task-field__task-list"
      />
      <button
        class="material-icons-outlined button button__add task-field__add-button"
      >
        add
      </button>
    </div>
  </div>
  <footer
    class="footer"
  >
    <a
      class="footer__link"
      href="https://github.com/palmerusaf/todo-list"
    >
      GitHub Repo
    </a>
    <img
      alt="git hub logo"
      class="footer__logo"
      src="https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg"
    />
  </footer>
</body>
`;

exports[`New project is updated correctly 1`] = `
<body>
  <div
    class="to-do-list"
  >
    <div
      class="project-field"
    >
      <button
        class="material-icons-outlined button button__arrow_back_ios project-field__expand-button"
      >
        arrow_back_ios
      </button>
      <span
        class="project-field__item"
        data-project-index="0"
      >
        <span
          class="project-field__title"
        >
          My Project
        </span>
        <span
          class="project-field__buttons"
        >
          <button
            class="material-icons-outlined button button__edit"
          >
            edit
          </button>
          <button
            class="material-icons-outlined button button__delete"
          >
            delete
          </button>
        </span>
      </span>
      <span
        class="project-field__item project-field__item--active"
        data-project-index="1"
      >
        <span
          class="project-field__title"
        >
          Changed Project
        </span>
        <span
          class="project-field__buttons"
        >
          <button
            class="material-icons-outlined button button__edit"
          >
            edit
          </button>
          <button
            class="material-icons-outlined button button__delete"
          >
            delete
          </button>
        </span>
      </span>
      <span
        class="project-field__item project-field__add"
      >
        <button
          class="material-icons-outlined button button__add"
        >
          add
        </button>
      </span>
    </div>
    <div
      class="task-field"
      data-project-index="1"
    >
      <h1
        class="task-field__project-title"
      >
        Task List for Changed Project
      </h1>
      <div
        class="task-field__task-list"
      />
      <button
        class="material-icons-outlined button button__add task-field__add-button"
      >
        add
      </button>
    </div>
  </div>
  <footer
    class="footer"
  >
    <a
      class="footer__link"
      href="https://github.com/palmerusaf/todo-list"
    >
      GitHub Repo
    </a>
    <img
      alt="git hub logo"
      class="footer__logo"
      src="https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg"
    />
  </footer>
</body>
`;

exports[`New task is created correctly 1`] = `
<body>
  <div
    class="to-do-list"
  >
    <div
      class="project-field"
    >
      <button
        class="material-icons-outlined button button__arrow_back_ios project-field__expand-button"
      >
        arrow_back_ios
      </button>
      <span
        class="project-field__item project-field__item--active"
        data-project-index="0"
      >
        <span
          class="project-field__title"
        >
          My Project
        </span>
        <span
          class="project-field__buttons"
        >
          <button
            class="material-icons-outlined button button__edit"
          >
            edit
          </button>
          <button
            class="material-icons-outlined button button__delete"
          >
            delete
          </button>
        </span>
      </span>
      <span
        class="project-field__item project-field__add"
      >
        <button
          class="material-icons-outlined button button__add"
        >
          add
        </button>
      </span>
    </div>
    <div
      class="task-field"
      data-project-index="0"
    >
      <h1
        class="task-field__project-title"
      >
        Task List for My Project
      </h1>
      <div
        class="task-field__task-list"
      >
        <span
          class="task-field__task-item"
          data-due-date="2020-11-09"
          data-task-index="0"
        >
          <div
            class="task-field__main-view"
          >
            <label
              class="task-field__label task-field__title"
            >
              Title: 
              <span>
                Task One Title
              </span>
            </label>
            <label
              class="task-field__label task-field__due-date"
            >
              Due date: 
              <span>
                over 1 year ago
              </span>
            </label>
            <button
              class="material-icons-outlined button button__arrow_back_ios task-field__expand-button"
            >
              arrow_back_ios
            </button>
          </div>
          <div
            class="task-field__expanded-view task-field__expanded-view--hidden"
          >
            <div
              class="task-field__details"
            >
              <label
                class="task-field__label task-field__description"
              >
                Description: 
                <span>
                  Task One Description
                </span>
              </label>
              <label
                class="task-field__label task-field__priority"
              >
                Priority: 
                <span>
                  Low
                </span>
              </label>
            </div>
            <div
              class="task-field__button-field"
            >
              <button
                class="material-icons-outlined button button__check"
              >
                check
              </button>
              <button
                class="material-icons-outlined button button__edit"
              >
                edit
              </button>
              <button
                class="material-icons-outlined button button__delete"
              >
                delete
              </button>
            </div>
          </div>
        </span>
      </div>
      <button
        class="material-icons-outlined button button__add task-field__add-button"
      >
        add
      </button>
    </div>
  </div>
  <footer
    class="footer"
  >
    <a
      class="footer__link"
      href="https://github.com/palmerusaf/todo-list"
    >
      GitHub Repo
    </a>
    <img
      alt="git hub logo"
      class="footer__logo"
      src="https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg"
    />
  </footer>
</body>
`;

exports[`Project is selected correctly 1`] = `
<body>
  <div
    class="to-do-list"
  >
    <div
      class="project-field"
    >
      <button
        class="material-icons-outlined button button__arrow_back_ios project-field__expand-button"
      >
        arrow_back_ios
      </button>
      <span
        class="project-field__item"
        data-project-index="0"
      >
        <span
          class="project-field__title"
        >
          My Project
        </span>
        <span
          class="project-field__buttons"
        >
          <button
            class="material-icons-outlined button button__edit"
          >
            edit
          </button>
          <button
            class="material-icons-outlined button button__delete"
          >
            delete
          </button>
        </span>
      </span>
      <span
        class="project-field__item project-field__item--active"
        data-project-index="1"
      >
        <span
          class="project-field__title"
        >
          Project one
        </span>
        <span
          class="project-field__buttons"
        >
          <button
            class="material-icons-outlined button button__edit"
          >
            edit
          </button>
          <button
            class="material-icons-outlined button button__delete"
          >
            delete
          </button>
        </span>
      </span>
      <span
        class="project-field__item project-field__add"
      >
        <button
          class="material-icons-outlined button button__add"
        >
          add
        </button>
      </span>
    </div>
    <div
      class="task-field"
      data-project-index="1"
    >
      <h1
        class="task-field__project-title"
      >
        Task List for Project one
      </h1>
      <div
        class="task-field__task-list"
      />
      <button
        class="material-icons-outlined button button__add task-field__add-button"
      >
        add
      </button>
    </div>
  </div>
  <footer
    class="footer"
  >
    <a
      class="footer__link"
      href="https://github.com/palmerusaf/todo-list"
    >
      GitHub Repo
    </a>
    <img
      alt="git hub logo"
      class="footer__logo"
      src="https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg"
    />
  </footer>
</body>
`;

exports[`todo list and footer renders correctly 1`] = `
<body>
  <div
    class="to-do-list"
  >
    <div
      class="project-field"
    >
      <button
        class="material-icons-outlined button button__arrow_back_ios project-field__expand-button"
      >
        arrow_back_ios
      </button>
      <span
        class="project-field__item project-field__item--active"
        data-project-index="0"
      >
        <span
          class="project-field__title"
        >
          My Project
        </span>
        <span
          class="project-field__buttons"
        >
          <button
            class="material-icons-outlined button button__edit"
          >
            edit
          </button>
          <button
            class="material-icons-outlined button button__delete"
          >
            delete
          </button>
        </span>
      </span>
      <span
        class="project-field__item project-field__add"
      >
        <button
          class="material-icons-outlined button button__add"
        >
          add
        </button>
      </span>
    </div>
    <div
      class="task-field"
      data-project-index="0"
    >
      <h1
        class="task-field__project-title"
      >
        Task List for My Project
      </h1>
      <div
        class="task-field__task-list"
      />
      <button
        class="material-icons-outlined button button__add task-field__add-button"
      >
        add
      </button>
    </div>
  </div>
  <footer
    class="footer"
  >
    <a
      class="footer__link"
      href="https://github.com/palmerusaf/todo-list"
    >
      GitHub Repo
    </a>
    <img
      alt="git hub logo"
      class="footer__logo"
      src="https://upload.wikimedia.org/wikipedia/commons/9/91/Octicons-mark-github.svg"
    />
  </footer>
</body>
`;
